pyfstat/make_sfts.py

```diff
@@ -943,6 +941,175 @@ def __init__(
         )
 
 
+class TransientLineWriter(Writer):
+    """Inject a transient line into some data.
+
+    A transient line is defined as a constant amplitude and frequency excess power lasting
+    for a (not necesarilly) finite time span of an observing run.
+
+    In practice, it corresponds to a CW without Doppler or antenna-patern-induced amplitude modulation.
+
+    This uses MFD_v4
+    """
+
+    required_mfd_line_parameters = [
+        "Freq",
+        "phi0",
+        "h0",
+        "cosi",
+        "transientWindowType",
+        "transientStartTime",
+        "transientTau",
+    ]
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+
+        if self.detectors is None:
+            raise ValueError("MakeFakeData_v4 requires detector name to be given")
+        elif len(self.detectors.split(",")) > 1:
+            raise NotImplementedError(
+                "MakeFakeData_v4 does not support more than one detector at a time. "
+                "Please, do it properly"
+            )
+
+    def correct_line_amplitude(self, use_effective_h0, scale_by_duration):
```

Comment:
I don't like that name, there's nothing "incorrect" about just a fixed h0... `rescale_amplitude`?

Reply:
Also this seems a bit clunky as an interface, doing two completely disjoint operations, and in both cases assuming the user had already made a "good" guess of h0 in some other sense. It's also not called automatically by the class apparently?

Suggestions:
1. drop this and make it the user's responsibility to know which h0 they want
2. only drop the more trivial duration path, and add an optional `h0_eff` argument to the class itself which will at init time use that one and cosi to override h0

Reply:
a) `correct` should be understood in the sense of  "add extra terms to whatever you already have". I agree is not the best term, I just put it there to have it done, I'll change it once we clarify how to implement this.

b) I don't like to re-implement a function to multiply `h0` by `eta⁴ + 6 eta² + 1` every time, or another one related to the SNR scaling (since it's very easy to mess it up). I think we should have these relations somewhere (specially the h0_eff one).

The intended use case was to allow the user to apply at its own will either (or both) of these corrections. I.e. once the writer is set up, the user can choose whether to modify it or not.

Since `cosi` is completely ignored by MFDv4 in this case, I also though about using `h0eff` always if `cosi` is given.

So, a second proposal

- [ ] Use **always** `h0eff = h0 * (eta⁴ + 6 eta² + 1)`, which reduces to `h0` when `cosi=0` (default case).


As for re-scaling according to the time duration... we could keep this method only to do that?

Reply:
I find it confusing that this is supposed to be called after initialising the class. Can you clarify your intended use case? One makes a single instance, then reuses it for multiple injections with different cosi/duration but fixed effective strength...?

Reply:
> then reuses it for multiple injections with different cosi/duration but fixed effective strength

No, no re-use. 

One makes a single instance with the intention of, say, using h0_eff instead of h0.

Instead of computing somewhere `h0_eff`,  one gives `cosi` and calls this method to make the class handle the conversion to `h0_eff`.

Reply:
Then let's just use your `Use always h0eff = h0 * (eta⁴ + 6 eta² + 1), which reduces to h0 when cosi=0 (default case).` in `__init__`, but best with a comment in the docstring and a warning thrown that this is happening in the actual cosi!=0 case.

The `scale_by_duration` I'd really leave to the user, this is not a well-defined input behaviour.

Reply:
See 24246e6